FindClangFormat.cmake

```diff
@@ -0,0 +1,30 @@
+#
+#.rst:
+# FindClangFormat
+# ---------------
+#
+# The module defines the following variables
+#
+# ``CLANG_FORMAT_EXECUTABLE``
+#   Path to clang-format executable
+# ``CLANG_FORMAT_FOUND``
+#   True if the clang-format executable was found.
+#
+# Example usage:
+#
+# .. code-block:: cmake
+#
+#    find_package(ClangFormat)
+#    if(CLANG_FORMAT_FOUND)
+#      message("clang-format executable found: ${CLANG_FORMAT_EXECUTABLE}")
+#    endif()
+
+find_program(CLANG_FORMAT_EXECUTABLE
+  NAMES clang-format clang-format-3.9 clang-format-3.8
```

Comment:
What about older or newer versions of `clang-format`? `clang-format-4.0` already exists `clang-format-5.0` will come out soon, I believe `clang-3.3` was the first version in which `clang-format` was included. Ideally this would be forward compatible too, clang versioning will bump only the major version number from now on.